packages/transaction-controller/src/TransactionController.ts

```diff
@@ -982,9 +982,19 @@ export class TransactionController extends BaseController<
 
         /* istanbul ignore next */
         if (txObj?.blockNumber) {
-          meta.status = TransactionStatus.confirmed;
-          this.hub.emit(`${meta.id}:confirmed`, meta);
-          return [meta, true];
+          // transactions can be added to a block and still fail, so we need to check the transaction status before emitting the confirmed event
+          const verifyTransactionStatus =
+            await this.checkTxReceiptStatusIsFailed(transactionHash);
```

Comment:
Variable name `verifyTransactionStatus` is not intuitive.

Reply:
I updated this to `checkIfTxStatusIsFailed`. Does this work?

Reply:
Updated. https://github.com/MetaMask/core/pull/1773/commits/0f49dd6f37110c2f6755d07cc50a9de47015b3af